src/Umbraco.Core/Configuration/Models/ContentErrorPage.cs

```diff
@@ -1,15 +1,30 @@
 ﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Umbraco.Core.Configuration.Models.Validation;
 
 namespace Umbraco.Core.Configuration.Models
 {
-    public class ContentErrorPage
+    public class ContentErrorPage : ValidatableEntryBase
     {
-        //TODO introduce validation, to check only one of key/id/xPath is used.
-        public int ContentId { get; }
-        public Guid ContentKey { get; }
-        public string ContentXPath { get; }
-        public bool HasContentId { get; }
-        public bool HasContentKey { get; }
+        public int ContentId { get; set; }
+
+        public Guid ContentKey { get; set; }
+
+        public string ContentXPath { get; set; }
+
+        public bool HasContentId => ContentId != 0;
+
+        public bool HasContentKey => ContentKey != Guid.Empty;
+
+        public bool HasContentXPath => !string.IsNullOrEmpty(ContentXPath);
+
+        [Required]
         public string Culture { get; set; }
+
+        internal override bool IsValid()
+        {
+            return base.IsValid() &&
+                ((HasContentId ? 1 : 0) + (HasContentKey ? 1 : 0) + (HasContentXPath ? 1 : 0) == 1);
```

Comment:
So... We need exactly one of these states? 

Personally I would refactor to only ever allow one type of configuration. Too many options can be confusing.

Reply:
That seems to be the current expectation yes (have implemented this from a `TODO`).  Would suggest that refactor if done should be outside of this PR (and I guess there's potential that different solutions will have used different "identifiers", and as such we should support them all).